Sources/ClientRuntime/Retries/RetryStrategy.swift

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// `RetryStrategy` defines the interface for a type that controls the operation's retry behavior.
+/// > Note: Based on _The Design and Implementation of the AWS SDKs_ section 3.2.1, The Smithy Retry Interface, `RetryStrategy`.
+public protocol RetryStrategy {
```

Comment:
Can we make `RetryToken` be an associated type on the protocol? `RetryToken` will have different parameters depending on the implementation. Such as it will have `current retry cost` when used in a "bucket" concept such as one defined in the new-retries SEP